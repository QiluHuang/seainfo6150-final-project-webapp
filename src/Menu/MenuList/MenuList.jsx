import React, { useEffect, useState } from 'react'
import PropTypes from "prop-types";
import styles from "./MenuList.module.css";
import MenuListItem from "../MenuListItem/MenuListItem.jsx";
import { isEmpty } from "lodash";

import CategoryPageItem from '../../CategoryPage/CategoryPageItem/CategoryPageItem';
import SideBar from '../SideBar/SideBar';


const DishList = () => {
  const [fetchedData, setFetchedData] = useState([]);

  useEffect(() => {
    const fetchData = async () => {
      // performs a GET request
      const response = await fetch("https://demo1273150.mockable.io/cover");
      const responseJson = await response.json();
      setFetchedData(Object.values(responseJson));
    };

    if (!fetchedData.length) {
      fetchData();
    }
  }, [fetchedData]);

  let displayContent;

  console.log(fetchedData);

  if (fetchedData.length) {
    console.log(fetchedData);

    displayContent = (
      <ul className={styles.container}>
        {/* {Object.keys(props.dishes).map((dish) => ( */}

        {fetchedData.map((dish) => (
          <MenuListItem dish={dish} key={dish.slug} />))}
          {/* <CategoryPageItem dish={dish} key={dish.slug} />))} */}
      </ul>
    );
  } else {
    displayContent = <div>You have no data!</div>;
  }

  return (
    <div className={styles.canvas}>
      {displayContent}
    </div>
  );
}

// DishList.propTypes = {
//     dishes: PropTypes.array.isRequired
// }
export default DishList;
