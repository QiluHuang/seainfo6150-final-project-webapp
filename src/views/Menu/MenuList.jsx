import React from 'react';
import { Switch, Route, Link, useRouteMatch } from "react-router-dom";
import PropTypes from "prop-types";
import styles from "./MenuList.module.css";

import MenuListItem from './MenuListItem.jsx';



const MenuList = (props) => {
    console.log(props);
    return ( 
        
        <div className={styles.menuList_wrapper}>
            <div className={styles.title}>
                <h3> MENU </h3>
            </div>
            <div className={styles.menuList_container}>
                {props.menuList.map((dish) =>  (
                    <MenuListItem dish={dish} key={dish.slug} />
                ))}
            </div>
        </div>
    );
}

MenuList.propTypes = {
    menuList: PropTypes.array.isRequired
}

export default MenuList;