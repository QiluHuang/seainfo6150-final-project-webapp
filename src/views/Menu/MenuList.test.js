import React from 'react';
import { render } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";

import MenuList from './MenuList';

const dish = [
   {
        slug: "0101-chicken-with-sweet-cream-corn-soup",
        name: "Chicken With Sweet Cream Corn Soup",
        categoryID : "soup",
        category: "Soup",
        cover: true,
        text: "It’s full of protein and veggies and it’s naturally gluten-free, and it is sooo comforting and delicious.ter.",
        ingredients: "Chicken, Corn, Ground ginger and garlic powder, Green Onions, Carrots, Cornstarch, Eggs, Toasted sesame oil, Salt and pepper",
        image: {
            _url: "https://imageresizer.static9.net.au/4X0xzcYzOVZeMKP7lsvoNmdEHz4=/1274x717/smart/https%3A%2F%2Fprod.static9.net.au%2F_%2Fmedia%2FImages%2FKitchen%2F2011%2F09%2F22%2F11%2F34%2Fchickensoup.jpg",
            _height: "282",
            _width: "501"
        },
        price: "$9.99"
    }   
];

describe("MenuList tests", () => {
    it("renders correctly", () => {
      const { container } = render(<Router><MenuList menuList={dish}/></Router>);
      expect(container).toMatchSnapshot();
    });
});
  