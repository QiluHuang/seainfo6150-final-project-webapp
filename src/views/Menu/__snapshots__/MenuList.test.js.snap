// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuList tests renders correctly 1`] = `
<div>
  <div
    class="menuList_wrapper"
  >
    <div
      class="title"
    >
      <h3>
         MENU 
      </h3>
    </div>
    <div
      class="menuList_container"
    >
      <div
        class="container"
      >
        <a
          href="/menu/soup"
        >
          <div
            class="wrapper"
          >
            <div
              class="upper_image"
            >
              <div>
                <img
                  alt="Chicken With Sweet Cream Corn Soup"
                  class="img"
                  src="https://imageresizer.static9.net.au/4X0xzcYzOVZeMKP7lsvoNmdEHz4=/1274x717/smart/https%3A%2F%2Fprod.static9.net.au%2F_%2Fmedia%2FImages%2FKitchen%2F2011%2F09%2F22%2F11%2F34%2Fchickensoup.jpg"
                />
              </div>
            </div>
            <h5
              class="name"
            >
              Soup
            </h5>
          </div>
        </a>
      </div>
    </div>
  </div>
</div>
`;
